.changeset/dull-taxis-warn.md

```diff
@@ -0,0 +1,5 @@
+---
+'astro': patch
+---
+
+Adds deprecation warning on `getEntryBySlug` and `getDataEntryById` functions exported by `astro:content`.
```

Comment:
After thinking about it, maybe this would be better:
```suggestion
Deprecates `getEntryBySlug` and `getDataEntryById` functions exported by `astro:content` in favor of `getEntry`.
```
This would let users know directly that they should look for `getEntry`.

Reply:
Yes, I like this suggestion!